### ISSUE TITLE
Checkpoints continue to grow after the first restore

### ISSUE THREAD

**Issue by nbro** [tensorflow/tensorflow#16025]
### System information
- No

- **OS Platform and Distribution**: Mac OS X, v 10.13.2
- **TensorFlow installed from**: binary
- **TensorFlow version**: v1.3.0-rc1-5211-gab0fcac 1.5.0-dev20171126
- **Python version**: Python 3.5.0
- **Bazel version**: N/A
- **GCC/Compiler version**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: Intel Iris Pro 1536 MB
- **Exact command to reproduce**: N/A
- **Have I written custom code**: N/A

### Describe the problem

The parameter `max_to_keep` of the `Saver` class does not seem to have effect once a model and its training variables are restored. In other words, the first time I train my model, the saver is keeping only `max_to_keep` checkpoints. Then I interrupt the training. Later, when I resume it, the number of checkpoints keeps going without any apparent limit.

### Related issues

- https://github.com/tensorflow/tensorflow/issues/5929
- https://github.com/tensorflow/tensorflow/issues/6326

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code

**Comment by nbro** [comment]
@tensorflowbutler Ok, I have added it.

**Comment by nbro** [comment]
I've just noticed that the file `checkpoint`, which keeps track of the names of the last checkpoints, only contains the last `max_to_keep` filenames.

**Comment by tensorflowbutler** [comment]
The original poster has replied to this issue after the stat:awaiting response label was applied.

**Comment by reedwm** [comment]
I can reproduce with

```python
import tensorflow as tf

gs = tf.Variable(0, name='global_step')
inc_gs = gs.assign_add(1)
saver = tf.train.Saver([gs], max_to_keep=2)
with tf.Session() as sess:
  sess.run(tf.global_variables_initializer())
  for _ in range(3):
    saver.save(sess, '/tmp/ckpts/model.ckpt', gs)
    sess.run(inc_gs)

saver = tf.train.Saver([gs], max_to_keep=2)
with tf.Session() as sess:
  saver.restore(sess, tf.train.latest_checkpoint('/tmp/ckpts'))
  for _ in range(3):
    saver.save(sess, '/tmp/ckpts/model.ckpt', gs)
    sess.run(inc_gs)
```

@panyx0718, do you know if this is intended?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 100 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 116 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 131 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by bignamehyp** [comment]
Is it still an issue for the latest TF?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by bignamehyp** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by nbro** [comment]
@bignamehyp 

> Please update the issue when new information becomes available, and we will reopen the issue

Update the issue? Why should I need to update the issue? TF developers have to solve this issue. There's nothing to update.
